core/GlobalState.cc

```diff
@@ -697,11 +697,6 @@ ArgInfo &GlobalState::enterMethodArgumentSymbol(Loc loc, SymbolRef owner, NameRe
     ENFORCE(name.exists(), "entering symbol with non-existing name");
     SymbolData ownerScope = owner.dataAllowingNone(*this);
 
-    for (auto &arg : ownerScope->arguments()) {
```

Comment:
This makes this method dissimilar to other `enter`s.
Why was this neccessary? If it's necessary, can we rename the method to make it not have a similar name to others?

Reply:
This was deleted because it broke the assumption in this `ENFORCE`, which stipulates that the arguments list expands every time that we get a new method argument symbol. https://github.com/sorbet/sorbet/blob/a32794361584bab9f7af6e5729c530ce7311e86d/namer/namer.cc#L394-L396 Alternately, I could rewrite or delete the `ENFORCE` there, which I think is trying to maintain an invariant that's not necessary to maintain.

Reply:
TBH, I don't understand why this enforce was firing.
On the contrary, this seems to me like we're duplicating argument symbols in incremental runs, could you verify?

Reply:
I think implementing the following TODO is the proper way:
https://github.com/sorbet/sorbet/blob/a32794361584bab9f7af6e5729c530ce7311e86d/namer/namer.cc#L550